src/client/src/sync/controller.rs

```diff
@@ -137,55 +134,33 @@ where
     }
 
     // process updates from sc
+    #[instrument(
+        skip(self, updates),
+        fields(epoch = updates.epoch)
+    )]
     async fn process_updates(&mut self, updates: MetadataUpdate<S>) -> Result<(), IoError> {
-        //use fluvio_sc_schema::message::MsgType;
-        //use crate::metadata::store::actions::LSUpdate;
-
-        if !updates.all.is_empty() {
-            debug!(
-                "processing {}, sync all items: {}",
-                S::LABEL,
-                updates.all.len()
-            );
-            let mut objects: Vec<CacheMetadataStoreObject<S>> = vec![];
-            for meta in updates.all.into_iter() {
-                let store_obj: Result<CacheMetadataStoreObject<S>, _> = meta.try_into();
-                match store_obj {
-                    Ok(obj) => {
-                        objects.push(obj);
-                    }
-                    Err(err) => {
-                        return Err(IoError::new(
-                            ErrorKind::InvalidData,
-                            format!("problem converting: {}", err),
-                        ));
-                    }
-                }
-            }
-            self.store.store().sync_all(objects).await;
+        if updates.all.is_empty() {
+            debug!("Received no updates: skipping");
+            return Ok(());
         }
 
-        /*
-        TODO: Need to fix this, Need conversion from Metadata to LSUpdate
-        if !updates.changes.is_empty() {
-            debug!(
-                "processing {}, sync change items: {}",
-                S::LABEL,
-                updates.changes.len()
-            );
-
-            let changes = updates.changes.into_iter()
-                .map(|change| {
-
-                    match change.header {
-
-                        MsgType::UPDATE => LSUpdate::Mod(change.content),
-                        MsgType::DELETE => LSUpdate::Delete(change.content.key_owned()),
-
-                    }
-                }).collect();
+        debug!(count = updates.all.len(), "Syncing updates:");
```

Comment:
can you comment on what changes happen here?

Reply:
Sure. Previously, there was a really big if-statement like

```rust
if !updates.all.is_empty() {
    // ...
}
```

which was causing a lot of rightward drift. So all I did was remove that if-statement and shift everything inside back to the left, and at the top I added the opposite if-statement to short-circuit the function:

```rust
if updates.all.is_empty() {
    // ...
    return Ok(());
}
```

So it just made the code easier to read while I was trying to figure out what was going wrong.